src/wing/CCPACSWingCell.cpp

```diff
@@ -402,13 +404,85 @@ TopoDS_Shape CCPACSWingCell::GetSkinGeometry(TiglCoordinateSystem cs) const
     }
 }
 
+void CCPACSWingCell::TrimSpanwise(GeometryCache& cache,
+                                  SpanWiseBorder border,
+                                  CCPACSWingCellPositionSpanwise const& positioning,
+                                  double tol) const
+{
+    if (!positioning.GetSpanwiseContourCoordinate_choice1()){
+        throw CTiglError("Internal Error when trying to create the cell geometry");
+    }
+
+    double trim_coordinate = positioning.GetSpanwiseContourCoordinate_choice1().get();
+
+    for(auto row = cache.rgsurface.Root(); row; row = row->UNext()) {
+        for (auto current = row; current; current = current->VNext()) {
+            if (current->GetAnnotation().keep) {
+                if (trim_coordinate <= current->GetAnnotation().scmin) {
+                    if (border == SpanWiseBorder::Outer) {
+                        // remove this face from the rgsurface
+                        current->GetAnnotation().keep = false;
+                    }
+                }
+                else if (trim_coordinate >= current->GetAnnotation().scmax) {
+                    if (border == SpanWiseBorder::Inner) {
+                        // remove this face from the rgsurface
+                        current->GetAnnotation().keep = false;
+                    }
+                }
+                else {
+                    //trim the face
+
+                    double alpha = (trim_coordinate - current->GetAnnotation().scmin) / (current->GetAnnotation().scmax - current->GetAnnotation().scmin);
+                    double v = (1-alpha)*current->VMin() + alpha * current->VMax();
+
+                    double vmin = current->VMin();
+                    double vmax = current->VMax();
+                    bool trim = false;
+                    if (border == SpanWiseBorder::Inner) {
+                        if ( fabs(v - vmin) > tol ) {
+                            vmin = v;
+                            current->GetAnnotation().scmin = trim_coordinate;
+                            trim = true;
+                        }
+                    }
+                    else {
+                        if ( fabs(vmax - v) > tol ) {
+                            vmax = v;
+                            current->GetAnnotation().scmax = trim_coordinate;
+                            trim = true;
+                        }
+                    }
+
+                    if (trim) {
+                        current->ReplaceFace(TrimFace(current->GetFace(),
```

Comment:
**Important Note!** :  I looked at the implementation of `TrimFace`. Currently, this will only work, if the face was not trimmed before and uses more complicated face boundaries (e.g. skewed edges, holes etc). 

We should create an issue to improve "TrimFace" to handle the general case as well.

Reply:
See #777